sklearn/ensemble/ensemble_selection.py

```diff
@@ -0,0 +1,261 @@
+        'accuracy': _accuracy,
+    }
+
+    def __init__(self, estimators=None, max_bag_estimators=50, bag_fraction=0.5,
+                 n_bags=20, n_best=1, score_metric='accuracy',
+                 prune_fraction=0, random_state=None, verbose=False):
+        self.estimators = estimators
+        self.named_estimators = dict(estimators)
+        self.max_bag_estimators = max_bag_estimators
+        self.bag_fraction = bag_fraction
+        self.n_bags = n_bags
+        self.n_best = n_best
+        self.score_metric = score_metric
+        self.prune_fraction = prune_fraction
+        self.random_state = random_state
+        self.verbose = verbose
+
+    def fit(self, X, y):
+        """ Conduct ensemble selection on the validation set (X, y).
+
+        Parameters
+        ----------
+        X : {array-like, sparse matrix}, shape = [n_samples, n_features]
+            Validation vectors, where n_samples is the number of samples and
+            n_features is the number of features.
+
+        y : array-like, shape = [n_samples]
+            Target values.
+
+        Returns
+        -------
+        self : object
+        """
+        if isinstance(y, np.ndarray) and len(y.shape) > 1 and y.shape[1] > 1:
+            raise NotImplementedError('Multilabel and multi-output'
+                                      ' classification is not supported.')
+
+        metric_names = self._metrics.keys()
+        if self.score_metric not in metric_names:
+            raise ValueError('score_metric %s not in %s' %
+                             (self.score_metric, metric_names))
+
+        if self.estimators is None or len(self.estimators) == 0:
+            raise AttributeError('Invalid `estimators` attribute, `estimators`'
+                                 ' should be a list of'
+                                 ' (string, fitted_estimator) tuples')
+
+        self.score_metric_ = self._metrics[self.score_metric]
+
+        # store each clf's score and predict_proba result on validation set
+        self.clfs_score_ = {}
+        self.clfs_proba_ = {}
+        for (name, clf) in self.estimators:
+            score, proba = self._get_score_of_model(X, y, clf)
+            self.clfs_score_[name] = score
+            self.clfs_proba_[name] = proba
+
+        self.classes_ = np.unique(y)
+        self.n_classes_ = len(self.classes_)
+
+        self._get_final_ensemble(X, y)
+
+    def predict(self, X):
+        n_clfs = sum(self.ensemble_.values())
+        ens_proba = np.zeros((X.shape[0], self.n_classes_))
+
+        for estimator_name in self.ensemble_:
+            weight = self.ensemble_[estimator_name]
+            proba = self.named_estimators[estimator_name].predict_proba(X)
+            ens_proba += proba * weight
+
+        ens_proba /= n_clfs
+        return np.argmax(ens_proba, axis=1)
+
+    # TODO:
+    #def predict_proba(self, X):
+
+
+    def _get_score_of_model(self, X, y, estimator):
+        proba = estimator.predict_proba(X)
```

Comment:
I use `predict_proba` here because we can store it and output it if user want to call ensemble selection model's `predict_proba`.